dotnet/src/SemanticKernel/Memory/ISemanticTextMemory.cs

```diff
@@ -14,54 +14,61 @@ public interface ISemanticTextMemory
     /// <summary>
     /// Save some information into the semantic memory, keeping a copy of the source information.
     /// </summary>
-    /// <param name="collection">Collection where to save the information</param>
-    /// <param name="id">Unique identifier</param>
-    /// <param name="text">Information to save</param>
-    /// <param name="description">Optional description</param>
-    /// <param name="cancel">Cancellation token</param>
-    public Task SaveInformationAsync(
+    /// <param name="collection">Collection where to save the information.</param>
+    /// <param name="id">Unique identifier.</param>
+    /// <param name="text">Information to save.</param>
+    /// <param name="description">Optional description.</param>
+    /// <param name="valueString">Optional value string for saving custom metadata.</param>
+    /// <param name="cancel">Cancellation token.</param>
+    /// <returns>Unique identifier of the saved memory record.</returns>
+    public Task<string> SaveInformationAsync(
         string collection,
         string text,
         string id,
         string? description = null,
+        string? valueString = null,
```

Comment:
valueString <== sounds to me like the "actual value" of the thing being stored. Perhaps metadataString? Or additionalContext?

Dare I ask: should this be a string or a Dict<string, string>?

Reply:
Happy to rename it to customString, or additionalMetadata.

I do not want to make this a dictionary. The next best option I think is to make a dictionary<string, object> as we shouldn't limit the schema of the metadata. It should be up to the user whether they want to serialize an object, or have a  simple string, or nothing at all for this field. However, when you start adding dictionaries, that's additional steps to add fields, serialize, and deserialize individual entries, and there are no guarantees that they can be properly serialized/deserialized at the storage level.

Reply:
renamed to additionalMetadata